synapse/handlers/search.py

```diff
@@ -238,8 +238,6 @@ async def search(
 
             results = search_result["results"]
 
-            results_map = {r["event"].event_id: r for r in results}
-
```

Comment:
ugh. If this function wasn't 50000 lines long, we'd be able to follow this sort of thing much more easily.

Reply:
Yeah, PyCharm noticed it and declared it unused. 😢 

I can refactor this method a bit first if you'd like. I think removing the `context` calculation would be easy enough.

Reply:
it would certainly be nice to refactor it if you have a few tuits. Suggest doing as a followup though.

Reply:
I took a look at doing this and it quickly spiraled. Will do as a follow-up.